engine/include/Engine/Core/Rendering/Renderer/IRenderer.hpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include "Engine/Core/Rendering/Window/IWindow.hpp"
+#include <memory> //std::unique_ptr
+
+namespace Engine::Core::Renderering
+{
+class IRenderer
+{
+  protected:
+    std::unique_ptr<IWindow> m_pWindow = nullptr;
+
+  protected:
+    virtual void initialize(const WindowCreateArg &arg) noexcept = 0;
+    virtual void release() noexcept = 0;
+
+  public:
+
+    [[nodiscard]] inline const IWindow const *getWindow() const noexcept
```

Comment:
I have a doubt: did we say we were putting all implementations outside the class declaration, or did we give exceptions for short methods?

I would rather get any kind of implementation out of the class declaration for clarity.

Reply:
I think that create inl juste for 1 function is maybe too much restrictive and don't keep it simple. To define in vocal